extensions/objects/models/objects.py

```diff
@@ -752,99 +752,6 @@ def normalize(cls, raw):
-                            in the expression.
-                        arguments: list(str). A list of arguments.
-                        dummies: list. A list of dummy values.
-                results: list(dict(str, *)). The list containing the final
-                    results of the required proof in the dict format
-                    containing following key-value pairs:
-                        top_kind_name: str. The top kind name in the
-                            expression.
-                        top_operator_name: str. The top operator name
-                            in the expression.
-                        arguments: list(str). A list of arguments.
-                        dummies: list. A list of dummy values.
-                default_proof_string: str. The default proof string.
-
-        Raises:
-            TypeError. Cannot convert to LogicQuestion schema.
-        """
-
-        def _validate_expression(expression):
-            """Validates the given expression.
-
-            Args:
-                expression: dict(str, *). The expression to be verified in the
-                    dict format.
-
-            Raises:
-                AssertionError. The specified expression is not in the correct
-                    format.
-            """
-            assert isinstance(expression, dict)
-            assert isinstance(
-                expression['top_kind_name'], python_utils.BASESTRING)
-            top_operator_name_type = (
-                int
-                if (
-                    expression['top_kind_name'] == 'constant' and
-                    'type' in expression and
-                    expression['type'] == 'integer'
-                ) else python_utils.BASESTRING
-            )
-            assert isinstance(
-                expression['top_operator_name'], top_operator_name_type)
-            _validate_expression_array(expression['arguments'])
-            _validate_expression_array(expression['dummies'])
-
-        def _validate_expression_array(array):
-            """Validates the given expression array.
-
-            Args:
-                array: list(dict(str, *)). The expression array to be verified.
-
-            Raises:
-                AssertionError. The specified expression array is not in the
-                    list format.
-            """
-            assert isinstance(array, list)
-            for item in array:
-                _validate_expression(item)
-
-        try:
-            assert isinstance(raw, dict)
-            _validate_expression_array(raw['assumptions'])
-            _validate_expression_array(raw['results'])
-            assert isinstance(
-                raw['default_proof_string'], python_utils.BASESTRING)
-
-            return copy.deepcopy(raw)
-        except Exception:
-            raise TypeError('Cannot convert to a logic question %s' % raw)
-
-
 class LogicErrorCategory(BaseObject):
```

Comment:
This should be removed too!

Reply:
And all other mentions of the logic error in the whole codebase.

Reply:
Done!